mage_ai/data_cleaner/column_types/column_type_detector.py

```diff
@@ -172,7 +172,13 @@ def infer_object_type(series, column_name, kwargs):
             ):
                 return ColumnType.ZIP_CODE
             else:
-                return ColumnType.NUMBER
+                clean_series = clean_series.str.replace(r'\.0*', '')
+                try:
+                    clean_series.astype(int)
+                    return ColumnType.NUMBER
+                except OverflowError:
+                    return ColumnType.CATEGORY_HIGH_CARDINALITY
```

Comment:
Logic for determining categorical feature is in the branch below. You can reuse some of the code.
```
        if clean_series_nunique / length >= 0.8:
            return ColumnType.TEXT

        if clean_series_nunique <= kwargs.get('category_cardinality_threshold', 255):
            return ColumnType.CATEGORY
        else:
            return ColumnType.CATEGORY_HIGH_CARDINALITY
```

Reply:
Fixed. I initially didn't perform this check on purpose since I assumed that it wouldn't be likely that an integer style string would occur frequently, but I realize  now that especially with user IDs there can actually be a lower cardinality set of integer-like categories.